activerecord/lib/active_record/connection_handling.rb

```diff
@@ -355,11 +355,11 @@ def clear_on_handler(handler)
       def resolve_config_for_connection(config_or_env)
         raise "Anonymous class is not allowed." unless name
 
-        owner_name = primary_class? ? Base.name : name
-        self.connection_specification_name = owner_name
+        owner = primary_class? ? Base : self
```

Comment:
On the next line, we now set `connection_specification_name` on `owner` instead of `self`. This means that `ApplicationRecord.connects_to` no longer changes `ApplicationRecord.connection_specification_name` from nil to "ActiveRecord::Base".